Linking error on Raspberry Pi 4

Hi,

I was trying to use compile pistache on Raspberry Pi 4 Model B and I found out that it fails when it is compiling a test:

`
[ 93%] Linking CXX executable run_request_size_test
/usr/bin/ld: ../src/libpistache.a(client.cc.o): in function 'Pistache::Http::Client::doRequest(Pistache::Http::Request)':
client.cc:(.text+0x6aec): undefined reference to '__atomic_fetch_add_8'
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/run_threadname_test.dir/build.make:88: tests/run_threadname_test] Error 1
make[1]: *** [CMakeFiles/Makefile2:1688: tests/CMakeFiles/run_threadname_test.dir/all] Error 2
/usr/bin/ld: ../src/libpistache.a(client.cc.o): in function 'Pistache::Http::Client::doRequest(Pistache::Http::Request)':
client.cc:(.text+0x6aec): undefined reference to '__atomic_fetch_add_8'
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/run_request_size_test.dir/build.make:88: tests/run_request_size_test] Error 1
make[1]: *** [CMakeFiles/Makefile2:2585: tests/CMakeFiles/run_request_size_test.dir/all] Error 2
[ 94%] Linking CXX executable run_headers_test
`

I think the pistache's CMakeLists.txt is not linking the `atomic` library.
I patched the file to include the missing call and it does work fine. If you are interested in the patch I can start a pull request.

Thanks!
Simone

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,6 +68,7 @@ if (PISTACHE_USE_SSL)
     link_libraries(-lssl -lcrypto)
 endif (PISTACHE_USE_SSL)
 
+link_libraries(-latomic)
 include_directories (${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 # Set version...
@@ -113,7 +114,7 @@ include_directories (${CMAKE_CURRENT_SOURCE_DIR}/include)
 # Set libraries...
 
     # Minimum...
-    set(LIBS "-lpistache -lpthread")
+    set(LIBS "-lpistache -lpthread -latomic")
 
     # If building with OpenSSL support...
     if(PISTACHE_USE_SSL)

